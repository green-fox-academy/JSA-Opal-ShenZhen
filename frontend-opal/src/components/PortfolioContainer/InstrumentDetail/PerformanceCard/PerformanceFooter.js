import React from 'react';
import PropTypes from 'prop-types';
import { CardItem, Button, Text } from 'native-base';
import styles from '../styles';

const IntervalFooter = ({ intervals }) => {
  const buttons = () =>
    intervals.map(item => (
      <Button transparent key={item.id}>
        <Text style={styles.footerText}>{item.val}</Text>
      </Button>
    ));

  return (
    <CardItem footer bordered style={styles.chartFooter}>
      {buttons(intervals)}
    </CardItem>
  );
};

IntervalFooter.propTypes = {
  intervals: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      val: PropTypes.string
    })
  ).isRequired
};

export default IntervalFooter;
