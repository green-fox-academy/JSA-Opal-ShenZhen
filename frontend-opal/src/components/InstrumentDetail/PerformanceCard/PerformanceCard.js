import React from 'react';
import { Content, Text, Card, CardItem, Spinner } from 'native-base';
import PropTypes from 'prop-types';
import { View } from 'react-native';

import LineChart from 'components/PortfolioContainer/LineChart';
import PerformanceFooter from './PerformanceFooter';
import styles from '../styles';

const PerformanceCard = ({ apiData }) => {
  const intervals = [
    { id: 1, val: '1D' },
    { id: 2, val: '2D' },
    { id: 3, val: '1W' },
    { id: 4, val: '1M' },
    { id: 5, val: '6M' },
    { id: 6, val: '1Y' },
    { id: 7, val: '5Y' }
  ];

  return (
    <CardItem padder style={styles.firstCard}>
      <View style={{ width: '100%' }}>
        <Card transparent>
          <Text style={styles.title}>Performance</Text>
        </Card>

        <Card>
          <View cardBody style={styles.detail}>
            {apiData.length === 0 ? (
              <View style={{ display: 'flex', justifyContent: 'center', alignItems: 'center' }}>
                <Spinner color="#e17055" />
              </View>
            ) : (
              <LineChart data={apiData} />
            )}
          </View>

          <PerformanceFooter intervals={intervals} />
        </Card>
      </View>
    </CardItem>
  );
};

PerformanceCard.defaultProps = {
  apiData: []
};

PerformanceCard.propTypes = {
  apiData: PropTypes.arrayOf(
    PropTypes.shape({
      x: PropTypes.string,
      y: PropTypes.number
    })
  )
};

export default PerformanceCard;
