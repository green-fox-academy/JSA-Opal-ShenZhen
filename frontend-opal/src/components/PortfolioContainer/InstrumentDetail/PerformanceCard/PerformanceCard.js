import React from 'react';
import { View } from 'react-native';
import { Content, Text, Card, CardItem } from 'native-base';

import LineChart from '../LineChart';
import PerformanceFooter from './PerformanceFooter';
import styles from '../styles';

const PerformanceCard = () => {
  const intervals = [
    { id: 1, val: '1D' },
    { id: 2, val: '2D' },
    { id: 3, val: '1W' },
    { id: 4, val: '1M' },
    { id: 5, val: '6M' },
    { id: 6, val: '1Y' },
    { id: 7, val: '5Y' }
  ];

  return (
    <Content padder style={styles.firstCard}>
      <Card transparent>
        <Text style={styles.title}>Performance</Text>
      </Card>

      <Card>
        <CardItem cardBody>
          <LineChart />
        </CardItem>

        <PerformanceFooter intervals={intervals} />
      </Card>
    </Content>
  );
};

export default PerformanceCard;
