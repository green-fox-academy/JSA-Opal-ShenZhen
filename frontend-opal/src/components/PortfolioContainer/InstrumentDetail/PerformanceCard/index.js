import PerformanceCard from './PerformanceCard';

export default PerformanceCard;