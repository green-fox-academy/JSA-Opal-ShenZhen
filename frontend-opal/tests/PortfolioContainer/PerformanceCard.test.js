import React from 'react';
import renderer from 'react-test-renderer';
import PerformanceCard from 'components/PortfolioContainer/InstrumentDetail/PerformanceCard';

describe('PerformanceCard Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<PerformanceCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
