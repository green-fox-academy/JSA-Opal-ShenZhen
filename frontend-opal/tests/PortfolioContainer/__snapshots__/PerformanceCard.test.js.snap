// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PerformanceCard Component renders correctly 1`] = `
<View
  padder={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "borderBottomWidth": 0,
          "borderColor": "#ccc",
          "borderLeftWidth": 0,
          "borderRadius": 2,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "elevation": null,
          "flexWrap": "nowrap",
          "marginBottom": 5,
          "marginLeft": 2,
          "marginRight": 2,
          "marginTop": 5,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
      transparent={true}
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "black",
              "fontSize": 24,
            },
          ]
        }
        uppercase={false}
      >
        Performance
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 0.5,
          "borderColor": "#ccc",
          "borderLeftWidth": 0.5,
          "borderRadius": 2,
          "borderRightWidth": 0.5,
          "borderTopWidth": 0.5,
          "elevation": 3,
          "flexWrap": "nowrap",
          "marginBottom": 5,
          "marginLeft": 2,
          "marginRight": 2,
          "marginTop": 5,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 1.5,
        }
      }
    >
      <View
        cardBody={true}
        style={
          Object {
            "backgroundColor": "#DFF1FD",
            "display": "flex",
            "flexDirection": "row",
            "height": 200,
            "justifyContent": "space-evenly",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
            }
          }
        >
          <ActivityIndicator
            animating={true}
            color="#e17055"
            hidesWhenStopped={true}
            size="large"
            style={
              Object {
                "height": 80,
              }
            }
          />
        </View>
      </View>
      <View
        bordered={true}
        footer={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderBottomWidth": 0.5,
              "borderColor": "#ccc",
              "borderRadius": 2,
              "borderTopWidth": 0.5,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            },
            Object {
              "display": "flex",
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-around",
            },
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#007aff",
                  "fontFamily": "System",
                  "fontSize": 16.5,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                },
                Object {
                  "color": "black",
                },
              ]
            }
            uppercase={false}
          >
            1D
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#007aff",
                  "fontFamily": "System",
                  "fontSize": 16.5,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                },
                Object {
                  "color": "black",
                },
              ]
            }
            uppercase={false}
          >
            2D
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#007aff",
                  "fontFamily": "System",
                  "fontSize": 16.5,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                },
                Object {
                  "color": "black",
                },
              ]
            }
            uppercase={false}
          >
            1W
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#007aff",
                  "fontFamily": "System",
                  "fontSize": 16.5,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                },
                Object {
                  "color": "black",
                },
              ]
            }
            uppercase={false}
          >
            1M
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#007aff",
                  "fontFamily": "System",
                  "fontSize": 16.5,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                },
                Object {
                  "color": "black",
                },
              ]
            }
            uppercase={false}
          >
            6M
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#007aff",
                  "fontFamily": "System",
                  "fontSize": 16.5,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                },
                Object {
                  "color": "black",
                },
              ]
            }
            uppercase={false}
          >
            1Y
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#007aff",
                  "fontFamily": "System",
                  "fontSize": 16.5,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                },
                Object {
                  "color": "black",
                },
              ]
            }
            uppercase={false}
          >
            5Y
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
